zilliqa/src/pool.rs

```diff
@@ -1,33 +1,83 @@
+use crate::{
+    crypto::Hash,
+    state::Address,
+    transaction::{SignedTransaction, VerifiedTransaction},
+};
 use std::{
     cmp::Ordering,
     collections::{BTreeMap, BinaryHeap},
 };
+use tracing::*;
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+struct NoncedTxIndex {
+    pub address: Address,
+    pub nonce: u64,
+}
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+struct IntershardTxIndex {
+    pub source_shard: u64,
+    pub link_nonce: u64,
+}
 
-use crate::{crypto::Hash, state::Address, transaction::VerifiedTransaction};
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+enum TxIndex {
+    /// from_address, nonce (unique for that address)
+    Nonced(Address, u64),
+    /// source_shard, nonce (unique for the bridge from that shard)
+    Intershard(u64, u64),
+}
```

Comment:
Should we use the structs defined above here?
```suggestion
enum TxIndex {
    /// from_address, nonce (unique for that address)
    Nonced(NoncedTxIndex),
    /// source_shard, nonce (unique for the bridge from that shard)
    Intershard(IntershardTxIndex),
}
```

Reply:
Good catch - I went back and forth a few times on how I wanted to do this, and it seems this got accidentally left in.

I'm leaning towards actually just deleting the structs, since they're not used anywhere else, need no methods or other structure, it's only two fields which is easy to keep track of, and if we do use them then constructing a `TxIndex` inline becomes a lot more verbose and boilerplaty.

I'm gonna do that for now, let me know if you prefer otherwise.